Week-23/todo-ui/src/pages/AddTodo.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from "react";
+import styles from "./AddTodo.module.less";
+import axios from "axios";
+
+const AddTodo = () => {
+  const [todo, setTodo] = useState("");
+  const [allTodos, setAllTodos] = useState([]);
+  const [todoToUpdate, setTodoToUpdate] = useState(null);
+
+  const getTodo = async () => {
+    try {
+      const data = await axios.get(`http://localhost:3000/api`);
+      console.log("data", data.data);
+      setAllTodos(data.data);
+    } catch (error) {
+      console.error("Error getting todos:", error);
+    }
+  };
+
+  const handleAddTodo = async () => {
+    try {
+      if (todoToUpdate) {
+        // If todoToUpdate is set, update the existing todo
+        await axios.put(`http://localhost:3000/api/${todoToUpdate._id}`, {
+          todo,
+        });
+        setTodoToUpdate(null); // Reset the todoToUpdate state
+      } else {
+        // Otherwise, add a new todo
+        await axios.post("http://localhost:3000/api", {
+          todo,
+        });
+      }
```

Comment:
Are we supposed to upsert in the add method here? If so, rename the function so it’s clear. 
Also the api url is hardcoded.